storage/innobase/buf/buf0flu.cc

```diff
@@ -2191,10 +2191,15 @@ ATTRIBUTE_COLD void buf_flush_ahead(lsn_t lsn, bool furious) noexcept
     if (limit < lsn)
     {
       limit= lsn;
-      buf_pool.page_cleaner_set_idle(false);
-      pthread_cond_signal(&buf_pool.do_flush_list);
       if (furious)
+      {
         log_sys.set_check_for_checkpoint();
+      wake:
+        buf_pool.page_cleaner_set_idle(false);
+        pthread_cond_signal(&buf_pool.do_flush_list);
+      }
+      else if (buf_pool.page_cleaner_idle())
```

Comment:
We wake up page cleaner independently of `furious` value. The only condition which influence on the decision about page cleaner wake up is `buf_pool.page_cleaner_idle()`. I would propose to write it as the following:
```
if (furious)
  log_sys.set_check_for_checkpoint();
if (buf_pool.page_cleaner_idle()) {
  buf_pool.page_cleaner_set_idle(false);
   pthread_cond_signal(&buf_pool.do_flush_list);
}
```